video_analytics_example/main.cpp

```diff
@@ -1259,6 +1266,37 @@ int main(int argc, char *argv[])
 
         std::cout << "\t. Done preparing VPP In/ Out parameters." << std::endl;
 
+        /* LP size */
+        if (VD_LP_resize)
+        {
+            if ( (MFX_CODEC_AVC == DecParams.mfx.CodecId) && /* Only for AVC */
+                 (MFX_PICSTRUCT_PROGRESSIVE == DecParams.mfx.FrameInfo.PicStruct)) /* ...And only for progressive!*/
+            {   /* it is possible to use decoder's post-processing */
+                DecoderPostProcessing.In.CropX = 0;
+                DecoderPostProcessing.In.CropY = 0;
+                DecoderPostProcessing.In.CropW = DecParams.mfx.FrameInfo.CropW;
+                DecoderPostProcessing.In.CropH = DecParams.mfx.FrameInfo.CropH;
+
+                DecoderPostProcessing.Out.FourCC = DecParams.mfx.FrameInfo.FourCC;
+                DecoderPostProcessing.Out.ChromaFormat = DecParams.mfx.FrameInfo.ChromaFormat;
+                DecoderPostProcessing.Out.Width = MSDK_ALIGN16(VPPParams.vpp.Out.Width);
+                DecoderPostProcessing.Out.Height = MSDK_ALIGN16(VPPParams.vpp.Out.Height);
+                DecoderPostProcessing.Out.CropX = 0;
+                DecoderPostProcessing.Out.CropY = 0;
+                DecoderPostProcessing.Out.CropW = VPPParams.vpp.Out.CropW;
+                DecoderPostProcessing.Out.CropH = VPPParams.vpp.Out.CropH;
+
+                //DecParams.ExtParam = reinterpret_cast<mfxExtBuffer**>(&DecoderPostProcessing);
+                DecParams.ExtParam[0] = (mfxExtBuffer*)&(DecoderPostProcessing);
+                DecParams.NumExtParam = 1;
+                std::cout << "\t.Decoder's post-processing is used for resizing\n"<< std::endl;
+
+                /* need to correct VPP params: re-size done after decoding
+                 * So, VPP for CSC NV12->RGBP only */
+                VPPParams.vpp.In.CropW = VPPParams.vpp.Out.CropW;
+                VPPParams.vpp.In.CropH = VPPParams.vpp.Out.CropH;
```

Comment:
it is just a patch to configure the decoding to enable VD+SFC, but I didn't see changes to pass the VPP output surface to decoding? because it needs to return two kinds of surface, one is decoding output, the other one is resized surface.

Reply:
Hello Ben,
I have added new version of patch, please look.
>> but I didn't see changes to pass the VPP output surface to decoding?
Ok, done.

>> because it needs to return two kinds of surface, one is decoding output, the other one is resized surface.
Actually, this is not strictly required, because application may use big surface with small cropping for processing. For example, surface resolution is 1920x1088, but cropping area is 352x288 only. (of course memory usage become higher is this case)